pkg/recipes/terraform/config/providers/aws_test.go

```diff
@@ -45,24 +50,28 @@ type mockAWSCredentialsProvider struct {
 func newMockAWSCredentialsProvider() *mockAWSCredentialsProvider {
 	return &mockAWSCredentialsProvider{
 		testCredential: &ucp_credentials.AWSCredential{
-			AccessKeyID:     testAWSCredentials.AccessKeyID,
-			SecretAccessKey: testAWSCredentials.SecretAccessKey,
+			Kind: ucp_datamodel.AWSAccessKeyCredentialKind,
+			AccessKeyCredential: &ucp_datamodel.AWSAccessKeyCredentialProperties{
+				AccessKeyID:     testAWSCredentials.AccessKeyCredential.AccessKeyID,
+				SecretAccessKey: testAWSCredentials.AccessKeyCredential.SecretAccessKey,
+			},
 		},
 	}
 }
 
 // Fetch returns mock AWS credentials for testing. It takes in a context, planeName and name and returns
 // an AWSCredential or an error if the credentials are empty.
+// TODO: update as part of IRSA support in Terraform provider
```

Comment:
Is this something you meant to handle in this PR?

Reply:
No, I will add it as part of Terraform support for IRSA PR ( upcoming). I had to make some minimal changes just so I could compile with the changes to support the model.

Reply:
I think this answers my question here: https://github.com/radius-project/radius/pull/7739#discussion_r1678790947. I'd suggest returning an error instead of setting just one property which will indirectly result into an error. If this is just temporary and we are planning to complete it before release, then I'm fine with either approach as long as we add clear comment about it.

Reply:
I was thinking since we use it only in the case where the type IRSA credential we might be OK, but it makes sense to not take risk. let me see if I can remove this property as well, and just retain the changes that use the new type.

Reply:
I removed the todo, since partial terraform changes were causing confusion like you and will mentioned in another comment. will revisit it as part of terraform PR

Reply:
makes sense, thanks. I think the TODO didn't get added to the commit, you might need to re-add and push. :) Updates in `aws.go` are reflecting as expecting.